eclass/ninja-utils.eclass

```diff
@@ -35,6 +44,19 @@ _NINJA_UTILS_ECLASS=1
 
 inherit multiprocessing
 
+case "${NINJA}" in
+	ninja)
+		NINJA_DEPEND=">=dev-util/ninja-1.8.2"
+	;;
+	samu)
```

Comment:
If you let the user override it, you're violating the metadata cache immutability rule; i.e. if the users set `NINJA=samu` (which I suppose we want to allow), they will have different deps than users who don't.

Reply:
Scratch that. I see cmake is doing something like that anyway, so *shrug*.

Reply:
I was unhappy with that too, but I'm not yet familiar enough with the eclasses to improve this. My original patch didn't allow for setting `NINJA` in the ebuild itself, but that was suggested by @thesamesam on irc which is how I ended up here.